pkg/reconciler/ingress/controller.go

```diff
@@ -182,6 +183,9 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 		func(ns, name string) (*corev1.Service, error) {
 			return serviceInformer.Lister().Services(ns).Get(name)
 		},
+		func(name string) (*corev1.Namespace, error) {
+			return kubernetesClient.CoreV1().Namespaces().Get(context.TODO(), name, metav1.GetOptions{})
```

Comment:
Use the `ctx` from `NewController`, rather than `TODO`?

Reply:
maybe, though it's not the right context. Hypothetically the reconciler can create a cancelable context, call cancel which won't stop the inflight Get request. 

That said I changed the code to propagate the right context!

Thanks!

Reply:
I changed to use a informer, no NA anymore.